R/pit.R

```diff
@@ -0,0 +1,146 @@
+#' Probability integral transform
+#'
+#' The probability integral transform (PIT). LOO-PIT is achieved with weighted sample.
+#'
+#' @param x (draws) A [`draws_matrix`] object or one coercible to a
+#' `draws_matrix` object.
+
+#' @param y (observations) A 1D vector of observations. Each element of `y`
+#' corresponds to a column in `x`.
+
+#' @param loo_weights A [`draws_matrix`] object or one coercible to a
+#' `draws_matrix` object. `loo_weights`` should have the same dimensions as `x`.
+
+#' @param log Are the weights provided in `loo_weights` log-weights.
+
+#' @details The `pit()` function computes the probability integral transform of
+#'   `y` using the empirical cumulative distribution computed from the samples
+#'   of `x`. These samples can further be provided (log-)weights in
+#    `loo_weights`, which enables for example the computation of LOO-PITs.
+#'
+#'   If `y` and `x` are discrete, randomisation is used to obtain continuous PIT
+#'   values. (see, e.g., Czado, C., Gneiting, T., Held, L.: Predictive model
+#'   assessment for count data.  Biometrics 65(4), 1254–1261 (2009).)
+#'
+#' @return A numeric vector of length `length(y)` containing the PIT values.
+#'
+#' @export
+pit <- function(x, y, loo_weights = NULL, log = TRUE) UseMethod("pit")
+
+#' @rdname pit
+#' @export
+pit.default <- function(x, y, loo_weights = NULL, log = TRUE) {
```

Comment:
Are weights only valid if they come from loo? Or can they be, in theory, any kind of weights? If so, can we just call them weights?

Reply:
I have changed them to weights. The difference to the weights already available to draws objects in `.weights` is, that this function requires a weight for each draw and each variable.
We were discussing with @n-kall that in the presence of the per-draw `.weights` they should also be taken into account.

Reply:
Okay. Makes sense.

Reply:
This could be a future extension for the functionality of `pit()`.